ArchiSteamFarm/Bot.cs

```diff
@@ -1173,8 +1173,21 @@ public sealed class Bot : IDisposable {
 			// We must escape our message prior to sending it
 			message = Escape(message);
 
-			for (int i = 0; i < message.Length; i += maxMessageLength) {
-				string messagePart = message.Substring(i, Math.Min(maxMessageLength, message.Length - i));
+			int i = 0;
+			int partLength;
+			while (i < message.Length) {
+				if (message.Length - i > maxMessageLength) {
+					int lastNewLine = message.LastIndexOf('\n', i+maxMessageLength, maxMessageLength);
```

Comment:
`\n` is not portable, use `Environment.Newline`. You'll also need to take into account that `Environment.Newline` length is not always 1.

Reply:
Damn. I knew about this. Sorry.

Reply:
Also, I don't understand what you mean about `Environment.Newline` length. It can be not 1, indeed, but if I cut the string before newline - it should not give me any problems, or am I missing something?

Reply:
If you're not making use of the remaining portion of the string in this loop, then yes, it doesn't matter for you.